.github/workflows/contracts-deploy.yml

```diff
@@ -4,25 +4,55 @@ name: contracts-e2e-tests-and-deploy
 on:
   workflow_call:
 
-# DO NOT TOUCH THIS:
+# DO NOT TOUCH THIS !
+# cancelling workflow when contracts deployment is in progress will leave the environment in a bad state
 concurrency:
   group: ${{ github.workflow }}
```

Comment:
If we allow this to be called from pull request (do we?), then should not it be `${{ github.ref }}-${{ github.workflow }}`?

Reply:
no - it needs to be unique, **always** or two concurrent runs will mess up the contracts deployment

Reply:
Right, I see now.